tests/stream.rs

```diff
@@ -0,0 +1,274 @@
+#[cfg(feature = "enable_logging")]
```

Comment:
These tests are good. It would be nice to somehow factor in the code with `tests/lzma.rs` to avoid code duplication, and make sure that new tests are automatically picked up on both "normal" and "stream" modes.

I'm not sure how to best do that yet (having tests generic over some trait for example), so it shouldn't block this pull request, but any idea is appreciated of course.

Reply:
Good idea. I made the common test functions like `round_trip` test both the non-streaming and streaming implementations and got rid of the stream.rs test module.